examples/usertests/resources/CoreReactiveTest.java

```diff
@@ -0,0 +1,316 @@
+import org.junit.runner.RunWith;
+
+import io.smallrye.mutiny.Uni;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.VertxOptions;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.RunTestOnContext;
+import io.vertx.ext.unit.junit.Timeout;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+
+import static org.hibernate.reactive.util.impl.CompletionStages.loop;
+import static org.hibernate.reactive.util.impl.CompletionStages.voidFuture;
+
+/**
+ * Base class for unit tests that need a connection to the selected db and
+ * need to wait for the end of the task for the assertion.
+ * <p>
+ *     Uses the {@link RunTestOnContext} rule to guarantee that all tests
+ *     will run in the Vert.x event loop thread (by default, the tests using Vert.x unit will start
+ *     in the JUnit thread).
+ * </p>
+ * <p>
+ *     Contains several utility methods to make it easier to test Hibernate Reactive
+ *     using Vert.x unit.
+ * </p>
+ */
+@RunWith(VertxUnitRunner.class)
+public abstract class CoreReactiveTest {
+
+	public static SessionFactoryManager factoryManager = new SessionFactoryManager();
+
+	@Rule
+	public Timeout rule = Timeout.seconds( 5 * 60 );
+
+	@ClassRule
+	public static RunTestOnContext vertxContextRule = new RunTestOnContext( () -> {
+		VertxOptions options = new VertxOptions();
+		options.setBlockedThreadCheckInterval( 5 );
+		options.setBlockedThreadCheckIntervalUnit( TimeUnit.MINUTES );
+		Vertx vertx = Vertx.vertx( options );
+		return vertx;
+	} );
+
+	private Object session;
+
+	private ReactiveConnection connection;
+
+	protected static void test(TestContext context, CompletionStage<?> work) {
+		test( context.async(), context, work );
+	}
+
+	/**
+	 * For when we need to create the {@link Async} in advance
+	 */
+	protected static void test(Async async, TestContext context, CompletionStage<?> work) {
+		work.whenComplete( (res, err) -> {
+			if ( err != null ) {
+				context.fail( err );
+			}
+			else {
+				async.complete();
+			}
+		} );
+	}
+
+	protected static void test(TestContext context, Uni<?> uni) {
+		test( context.async(), context, uni );
+	}
+
+	/**
+	 * For when we need to create the {@link Async} in advance
+	 */
+	public static void test(Async async, TestContext context, Uni<?> uni) {
+		uni.subscribe().with(
+				res -> async.complete(),
+				throwable -> context.fail( throwable )
+		);
+	}
+
+	private static boolean doneTablespace;
+
+	protected Configuration constructConfiguration() {
+		Configuration configuration = new Configuration();
+		String jdbcURL = "jdbc:postgresql://localhost:5432/hreact?loggerLevel=OFF&user=hreact&password=hreact";
+		String dbType = "PostgreSQL";
+		configuration.setProperty( Settings.HBM2DDL_AUTO, "create" );
+		configuration.setProperty( "db", dbType );
+		configuration.setProperty( Settings.URL, jdbcURL );
+		if ( configuration.getProperty( "db" ).equals( "DB2" ) && !doneTablespace ) {
```

Comment:
I don't think this configuration for DB2 is needed most of the time, we could omit it.
It makes sense only for specific use cases